site/en/guides/porting/set-up-the-build-environment.md

```diff
@@ -1,122 +1,143 @@
+# TODO: This is just pseudocode at this point since the repo hasn't actually been created.
+~/repos/ot-efr32$ wget https://github.com/SiliconLabs/ot-stubbed-platform/archive/refs/heads/main.zip
+~/repos/ot-efr32$ unzip main.zip
+~/repos/ot-efr32$ ls -alh
+total 64K
+drwxr-xr-x 10 user user 4.0K Apr 13 16:17 .
+drwxr-xr-x 14 user user 4.0K Apr 16 16:46 ..
+drwxr-xr-x  4 user user 4.0K Apr 16 16:29 build
+-rw-r--r--  1 user user 3.2K Apr 12 18:19 .clang-format
+-rw-r--r--  1 user user 3.1K Apr 12 18:19 CMakeLists.txt
+drwxr-xr-x  3 user user 4.0K Apr 12 18:19 examples
+drwxr-xr-x  9 user user 4.0K Apr 13 16:49 .git
+drwxr-xr-x  3 user user 4.0K Apr 12 18:19 .github
+-rw-r--r--  1 user user 1.4K Apr 12 18:19 .gitignore
+-rw-r--r--  1 user user  234 Apr 12 18:19 .gitmodules
+-rw-r--r--  1 user user 1.5K Apr 12 18:19 LICENSE
+drwxr-xr-x 12 user user 4.0K Apr 13 16:49 openthread
+-rw-r--r--  1 user user 2.3K Apr 12 18:19 README.md
+drwxr-xr-x  2 user user 4.0K Apr 12 18:19 script
+drwxr-xr-x  7 user user 4.0K Apr 12 18:19 src
+drwxr-xr-x  4 user user 4.0K Apr 12 18:19 third_party
+```
 
+## Step 3: Add submodules
+
+The next step is to add [`openthread`](https://github.com/openthread/openthread) and any other required repos as submodules
+
+```shell
+~/repos/ot-efr32$ git submodule add git@github.com:openthread/openthread.git
+Cloning into '/home/user/repos/ot-efr32/openthread'...
+remote: Enumerating objects: 78281, done.
+remote: Counting objects: 100% (1056/1056), done.
+remote: Compressing objects: 100% (488/488), done.
+remote: Total 78281 (delta 639), reused 864 (delta 556), pack-reused 77225
+Receiving objects: 100% (78281/78281), 76.62 MiB | 35.24 MiB/s, done.
+Resolving deltas: 100% (61292/61292), done.
 ```
-case ${with_examples} in
- 
-    ...
- 
-    efr32)
-        OPENTHREAD_EXAMPLES_EFR32=1
-        AC_DEFINE_UNQUOTED([OPENTHREAD_EXAMPLES_EFR32],[${OPENTHREAD_EXAMPLES_EFR32}],[Define to 1 if you want to use efr32 examples])
-        ;;
- 
-...
- 
-esac
- 
-...
- 
-AC_SUBST(OPENTHREAD_EXAMPLES_EFR32)
-AM_CONDITIONAL([OPENTHREAD_EXAMPLES_EFR32], [test "${OPENTHREAD_EXAMPLES}" = "efr32"])
+
+> Note: Any non-original derivative code (for example, linker script or
+toolchain startup code) must be contained in the `third_party` directory.
+
+For this example, we'll be adding a lite version of the Silicon Labs Gecko SDK as a submodule.
+
+```shell
+~/repos/ot-efr32$ cd third_party
+~/repos/ot-efr32/third_party$ git submodule add git@github.com:SiliconLabs/sdk_support.git
+Cloning into '/home/user/repos/ot-efr32/third_party/sdk_support'...
+remote: Enumerating objects: 32867, done.
+remote: Counting objects: 100% (8181/8181), done.
+remote: Compressing objects: 100% (3098/3098), done.
+remote: Total 32867 (delta 4945), reused 7469 (delta 4732), pack-reused 24686
+Receiving objects: 100% (32867/32867), 128.83 MiB | 30.91 MiB/s, done.
+Resolving deltas: 100% (19797/19797), done.
 ```
 
-### Makefile output directory
+## Step 4: Modifying scripts
```

Comment:
Does including this section make the guide too verbose? There really isn't a description for these files anywhere

Reply:
I think we should describe the standard set of scripts and their purpose to encourage consistency across the repos.